src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -132,6 +133,21 @@ class IOUAmountPage extends React.Component {
         });
     }
 
+    /**
+     * Update amount on amount change
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit
+     *
+     * @param {String} amount
+     */
+    updateAmount(amount) {
+        // Regex to validate decimal number with up to 3 decimal numbers
```

Comment:
I think we can also DRY this up a bit by creating another function called `validateAmount`

```javascript
validateAmount(amount) {
    const decimalNumberRegex = new RegExp(/^\d+(\.\d{0,3})?$/, 'i');
    return amount === '' || new RegExp(/^\d+(\.\d{0,3})?$/, 'i').test(amount);
}

updateAmountNumberPad(key) {
    // Backspace button is pressed
    if (key === '<') {
        if (this.state.amount.length > 0) {
            this.setState(prevState => ({
                amount: prevState.amount.slice(0, -1),
            }));
        }
        return;
    }
    this.setState((prevState) => {
        // (review note) This might as well be called `amount` instead of `newValue`
        const amount = `${prevState.amount}${key}`;
        return this.validateAmount(amount) ? {amount} : prevState;
    });
}

updateAmount(amount) {
    if (this.validateAmount(amount)) {
        this.setState({amount});
    }
}
```

Reply:
Good catch!

Reply:
Tested android, web, mWeb again.